pydm/widgets/timeplot.py

```diff
@@ -231,24 +366,133 @@ def setCurves(self, new_list):
 
     curves = Property("QStringList", getCurves, setCurves)
 
+    def findCurve(self, pv_name):
+        for curve in self._curves:
+            if curve.address == pv_name:
+                return curve
+
+    def refreshCurve(self, curve):
+        """
+        Remove a curve currently being plotted on the timeplot, then redraw that curve, which could have been updated
+        with a new symbol, line style, line width, etc.
+        :param curve:
+        :return:
+        """
+        curve = self.findCurve(curve.channel)
+        if curve:
+            self.removeYChannel(curve)
+            self.addYChannel(y_channel=curve.address, color=curve.color, name=curve.address,
+                             lineStyle=curve.lineStyle, lineWidth=curve.lineWidth, symbol=curve.symbol,
+                             symbolSize=curve.symbolSize)
+
+    def annotateCurve(self, curve, annotation):
+        curve_point = CurvePoint(curve)
+        self.plotItem.addItem(curve_point)
+        annotation.setParentItem(curve_point)
+        curve_point.setPos(1)
+
+    def addLegendItem(self, item, pv_name, force_show_legend=False):
+        self._legend.addItem(item, pv_name)
+        self.setShowLegend(force_show_legend)
+
+    def removeLegendItem(self, pv_name):
+        self._legend.removeItem(pv_name)
+        if len(self._legend.items) == 0:
+            self.setShowLegend(False)
+
+    def add_annotation(self):
+        pass
+        # self.add_annot_mode = True
+        # self.mouse_right_click_proxy = SignalProxy(self.plotItem.scene().sigMouseClicked, rateLimit=60,
+        #                                            slot=self.mouseClicked)
+
+    def mouseClicked(self, evt):
+        pass
+        # if self.add_annot_mode and evt[0].button() == Qt.LeftButton:
```

Comment:
What are these commented code blocks?

Reply:
Do we need a `super()` here to pass to other `mouseClicked`?

Reply:
This is slot for a SignalProxy, so there's no inheritance relationship involved. A super() call won't be needed.

Reply:
A bunch of comments with a pass statement is not clear. Please add a note explaining why this method should just `return` and not just `pass` so it is clear for future work.